code/go/0chain.net/smartcontract/storagesc/allocation_eventdb.go

```diff
@@ -0,0 +1,239 @@
+package storagesc
+
+import (
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/state"
+	"0chain.net/core/common"
+	"0chain.net/smartcontract/dbs/event"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+)
+
+func allocationTableToStorageAllocation(alloc *event.Allocation, balances cstate.StateContextI) (*StorageAllocation, error) {
```

Comment:
Pass an `*event.EventDb` object instead of cstate.StateContextI object.

Reply:
done